src/utils.py

```diff
@@ -47,91 +72,112 @@ def read_path_if_provided(spark, path):
     if parquet_files:
         return spark.read.parquet(path)
 
-def add_delta(
-        df: pd.DataFrame,
-        metric: str,
-        previous_run: str,
-        latest_run: str
-    ):
-
-    df[f"Δ in number of {metric}"] = (
-        df[f"Nr of {metric} in {latest_run.split('-')[0]}"]
-        .sub(
-            df[f"Nr of {metric} in {previous_run.split('-')[0]}"],
-            axis=0, fill_value=0))
-
-def compare_entity(
-    df: pd.DataFrame,
-    entity_name: str,
-    latest_run: str,
-    previous_run: str
-) -> pd.DataFrame:
-
-    if entity_name in ['diseases', 'drugs', 'targets']:
+
+def add_delta(df: pd.DataFrame, metric: str, previous_run: str, latest_run: str):
+    """
+    It takes a dataframe, a metric, and two run names, and adds a column to the dataframe that shows the
+    difference in the number of that metric between the two runs
+
+    Args:
+      df (pd.DataFrame): the dataframe to add the delta column to
+      metric (str): the metric we're interested in
+      previous_run (str): the name of the previous run
+      latest_run (str): the latest run of the pipeline
+    """
+
+    df[f"Δ in number of {metric}"] = df[f"Nr of {metric} in {latest_run.split('-')[0]}"].sub(
+        df[f"Nr of {metric} in {previous_run.split('-')[0]}"], axis=0, fill_value=0
+    )
```

Comment:
It seems this function doesn't have return value.... Also I would suggest something like this:
```suggestion
    delta_column = f"Δ in number of {metric}"
    previous_column = f"Nr of {metric} in {latest_run.split('-')[0]}"
    latest_column = f"Nr of {metric} in {previous_run.split('-')[0]}"

    return df.assign(**{
        delta_column: lambda df: df[latest_column].sub(df[previous_column], axis=0, fill_value=0)
    })
```